DependencyInjection/DoctrineExtension.php

```diff
@@ -242,6 +243,19 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
             ManagerRegistryAwareConnectionProvider::class,
             new Definition(ManagerRegistryAwareConnectionProvider::class, [$container->getDefinition('doctrine')])
         );
+
+        $schemaManagerFactoryId = $connection['schema_manager_factory'];
+        if ($schemaManagerFactoryId === 'doctrine.dbal.legacy_schema_manager_factory') {
+            trigger_deprecation(
```

Comment:
any idea how to easily/centrally fix or silence this deprecation for the testsuite?

```
Remaining self deprecation notices (177)

  177x: Since doctrine/doctrine-bundle 2.9: Using "doctrine.dbal.legacy_schema_manager_factory" as schema_manager_factory is deprecated. Use "doctrine.dbal.default_schema_manager_factory" or a custom factory instead. See UPGRADE-2.9.md for details.
    6x in DoctrineExtensionTest::testAutomapping from Doctrine\Bundle\DoctrineBundle\Tests\DependencyInjection
    6x in DoctrineExtensionTest::testCacheConfiguration from Doctrine\Bundle\DoctrineBundle\Tests\DependencyInjection
    6x in XmlDoctrineExtensionTest::testDbalLoadFromXmlMultipleConnections from Doctrine\Bundle\DoctrineBundle\Tests\DependencyInjection
    6x in YamlDoctrineExtensionTest::testDbalLoadFromXmlMultipleConnections from Doctrine\Bundle\DoctrineBundle\Tests\DependencyInjection
    4x in MiddlewarePassTest::testAddMiddlewareWithExplicitTag from Doctrine\Bundle\DoctrineBundle\Tests\DependencyInjection\Compiler
```

I don't really want to change all the configuration fixtures and set `schema_manager_factory = doctrine.dbal.default_schema_manager_factory` :see_no_evil:

Reply:
Or is deprecating the service enough actually? :thinking:

Reply:
> I don't really want to change all the configuration fixtures and set `schema_manager_factory = doctrine.dbal.default_schema_manager_factory` 🙈

As soon as this bundle starts to support DBAL 4, this has to be done anyway. So why not do it now? (sorry)

Reply:
I don't see a need to trigger deprecation there actually. I don't see why users must specify schema manager explicitly.

Reply:
Ok so currently there is no default for this config. So it would be null. Which is deprecated in DBAL 3.6+. Also using the legacy factory is deprecated and not working with the upcoming DBAL 4.

So we should encourage users to use the default factory, right?

Reply:
Why not class_exists(LegacyFactory::class) ? LegacyFactory::class : DefaultFactory like I suggested? This way nothing breaks neither with DBAL 3 nor 4, deprecation will still be there but will come from DBAL and not doctrine bundle and users can get rid of it.

Reply:
Ok let me have a look at implementing it that way.

Reply:
@ostrolucky migrating from the legacy factory to the default factory is a BC break. And the DefaultFactory exists in DBAL 3.6 (while not being the default yet). We cannot do a class_exists check to use the default factory as that would break things for DBAL 3 users. We need to have an opt-in.

However, if DBAL 4 is used, the bundle should indeed default to using the DefaultFactory.

Reply:
class_exists(LegacyFactory::class) ? LegacyFactory::class : DefaultFactory is an opt in. It still defaults to legacyfactory so nothing changes.

Reply:
this is not an opt-in. It forces users of DBAL 3.6 to stay on the legacy behavior (which will trigger a deprecation warning) instead of being able to intentionally choose to upgrade to the new behavior (without waiting for the DBAL 4 upgrade itself).

Reply:
I was just talking about default value for `schema_manager_factory` bundle option. Option will still be there and users can change it to default factory